Update docs

sed-i opened the pull request:
Currently, [alertmanager docs on charmhub](https://charmhub.io/alertmanager-k8s/docs/) have the following structure:

- [Overview](https://charmhub.io/alertmanager-k8s)  (taken from README.md)
- [Docs - Overview](https://charmhub.io/alertmanager-k8s/docs)
- [Configuration](https://charmhub.io/alertmanager-k8s/docs/configuration)
- [Clustering](https://charmhub.io/alertmanager-k8s/docs/clustering)
- [Relations](https://charmhub.io/alertmanager-k8s/docs/relations)

All of the above, except Relations, is already incorporated in the README.md file.
I think the Relations doc can be entirely incorporated in metadata.yaml.

Fixes #109.

dstathis commented on README.md:
````diff
@@ -118,6 +118,10 @@ juju status alertmanager-k8s --format=json \
   | jq -r '.applications."alertmanager-k8s".units."alertmanager-k8s/0".address'
 ```
 
+So, if you navigate to these IPs you will get Alertmanager dashboard:
````
```suggestion
So, if you navigate to these IPs you will get the Alertmanager dashboard:
```

Abuelodelanada commented:
Why do you think the Relations doc can be omitted entirely?

Right now in Prometheus we already have Relations in README.md. From my point of view, we should have a a standardised way to write the README.md

Can we remove from this PR the modifications in `charm.py` and tests and create another PR with that?

sed-i commented:
> Why do you think the Relations doc can be omitted entirely?

Bad phrasing on my part. Rephrased.

> Right now in Prometheus we already have Relations in README.md.

After trimming down all the superfluous "any charm with X relation over the Y interface" statements, I wasn't left with much. Whatever left I moved into metadata.yaml.

> From my point of view, we should have a a standardised way to write the README.md

Agreed. I'm proposing to describe relations in-context (inside metadata.yaml).

> Can we remove from this PR the modifications in `charm.py` and tests and create another PR with that?

Good point. Split into #114
